app/services/control_rate_service.rb

```diff
@@ -11,6 +11,7 @@ def initialize(region, periods:, with_exclusions: false)
     @facilities = region.facilities
     @periods = periods
     @report_range = periods
+    @period_type = @report_range.begin.type
```

Comment:
Does it matter at all that you're using `begin` here and `end` below? I would guess all the periods are the same type, so probably not.

Reply:
Yup, they will be all the same type.